UPGRADE-3.0.md

```diff
@@ -102,11 +102,16 @@
 - Removed the `Event::getTagsContext()` method, use `Event::getTags()` instead
 - Removed the `Event::getUserContext()` method, use `Event::getUser()` instead
 - Renamed the `Event::getServerOsContext()` method to `Event::getOsContext()`
-- The signature of the `Scope::setUser()` method changed ot accept a plain array
+- The signature of the `Scope::setUser()` method changed to accept a plain array
 - Removed the `FlushableClientInterface` and `ClosableTransportInterface` interfaces. Their methods have been moved to the corresponding `ClientInterface` and `TransportInterface` interfaces
 - Removed the `Event::toArray()` and `Event::jsonSerialize()` methods
 - Removed the `Breadcrumb::toArray()` and `Breadcrumb::jsonSerialize()` methods
 - Removed the `Frame::toArray()` and `Frame::jsonSerialize()` methods
 - Removed the `Stacktrace::toArray()` and `Stacktrace::jsonSerialize()` methods
 - Removed the `SpoolTransport` class and the `SpoolInterface` interface with related implementation
 - Made the `Event::__construct()` method `private`, use the named constructors instead
+- The signature of `ClientInterface::captureEvent` changed to `ClientInterface::captureEvent(Event $event, ?EventHint $hint = null, ?Scope $scope = null)`
+- The signature of `HubInterface::captureEvent` changed to `HubInterface::captureEvent(Event $event, ?EventHint $hint = null)`
+- The signature of `captureEvent` changed to `captureEvent(Event $event, ?EventHint $hint = null)`
+- The signature of `Scope::applyToEvent` changed to `Scope::applyToEvent(Event $event)`
+- Global and scope event processors will no longer receive a second argument, callable should now have the signature `callable(Event $event)`
```

Comment:
What's the reason of this change? The second argument is the only way to pass additional data from the location where an event has been captured down to the event processors and I see no reason to drop it. Also we had at least one issue where this saved us, so :-1: on removing it

Reply:
Because there is no longer that second argument since we only pass around Event objects now. So there is no "payload" anymore?

Reply:
We could pass the new $hint instead of the $payload I guess?

Reply:
Yea, this is what it's done in the other SDKs